### ISSUE TITLE
No feedback when changing note lengths using screen readers

### ISSUE THREAD

**Issue by ptorpey** [musescore/MuseScore#16909]
### Issue type

Other type of issue

### Bug description

I am using NVDA and JAWS screen reders on Windows.

When changing the note lengths by hitting the numbers 1 through 6, there is no spoken feedback to the user using a screen reader.  The note lengths are changed according to which number key is pressed as they should be, but the user will not know until they begin entering notes.

This lack of feedback happens with both NVDA and JAWS.  Would be nice if there was some spoken feedback when these actions are taken.

--Pete

### Steps to reproduce

When entering notes, hit one of the number keys on the top row of the keyboard to change the note duration.  Problem: There is no speech feedback to the screen reader user when these actions are taken.

### MuseScore Version

latest

### Regression

I don't know

### Operating system

Windows 10

**Comment by shoogle** [comment]
For example:

User presses | We should hear
:---:|:---:
4 | "Eighth note"
5 | "Quarter note"
6 | "Half note"
Etc. | Etc.

This is another case where we want the screen reader to say something not related to focus. There was a similar problem in #12966 which @Eism solved in PR #13383 by changing the name of the focus element to match the desired text and then asking the screen reader to say the name again. That solution would probably work here too.

Also opening this up to the community in case anybody wants to attempt it.

**Comment by Ryan17Lewis** [comment]
@cbjeukendrup Would you mind if I give this issue a shot?

**Comment by cbjeukendrup** [comment]
@Ryan17Lewis Feel free to do so!

**Comment by shubham-shinde-442** [comment]
I would like to work on this issue .

**Comment by shoogle** [comment]
@Ryan17Lewis, are you still working on this?

**Comment by shubham-shinde-442** [comment]
@shoogle I think Ryan is not active, can I work on this?

**Comment by cbjeukendrup** [comment]
I'd say, go ahead; after a month of inactivity, that seems reasonable to me.

**Comment by shoogle** [comment]
@shubham-shinde-442 asked for some hints with fixing this.

PR https://github.com/musescore/MuseScore/pull/13383 implemented a new function [`NotationAccessibility::setTriggeredCommand()`](https://github.com/musescore/MuseScore/pull/13383/files#diff-dfbd03195bca420089a4ad24442e4acd3cd523ed56abbaad93e7050cb9c05c77R115-R123). I think all you need to do is [call](https://github.com/musescore/MuseScore/pull/13383/files#diff-8aba8e28a78395a6ee6e88aadf15634fe55fb9073261bfff381d8ac3d7af07baR656-R657) this function with the [string](https://github.com/musescore/MuseScore/issues/16909#issuecomment-1476344913) you want to be read (i.e. "Quarter note", etc.).

If you [search](https://github.com/musescore/MuseScore/wiki/Find-your-way-around-the-code) the source code for "Quarter note", you'll see this is associated with action code [`"pad-note-4"`](https://github.com/musescore/MuseScore/blob/78a60610c9b11742095e41de2aa16ea847a21e86/src/notation/internal/notationuiactions.cpp#L2019-L2025) and is assigned shortcut key 5 [in shortcuts.xml](https://github.com/musescore/MuseScore/blob/78a60610c9b11742095e41de2aa16ea847a21e86/src/framework/shortcuts/data/shortcuts.xml#L609-L613).

You need to look for other places where `"pad-note-4"` appears in the code. One of those places will be the place where its functionality is implemented (i.e. where the note input duration is changed to a quarter note). You need to find that place, and somewhere around there is where you need to call `NotationAccessibility::setTriggeredCommand("Quarter note")`.

**Comment by DmitryArefiev** [comment]
Implemented in #21957

@ptorpey Please try it on your side. I found some minor things (https://github.com/musescore/MuseScore/pull/21957#issuecomment-2020042607 ) If it's important (or if you find anything else) please log it separately

**Comment by ptorpey** [comment]
Works great.  I now get speech feedback when changing the note lengths with the number keys.  Thanks for improving this feedback!

 

--Pete
